packages/jest-config/src/__tests__/normalize-test.js

```diff
@@ -827,14 +827,35 @@ describe('preset', () => {
       {},
     );
 
-    expect(options.moduleNameMapper).toEqual([['b', 'b'], ['a', 'a']]);
+    expect(options.moduleNameMapper).toEqual([['a', 'a'], ['b', 'b']]);
     expect(options.modulePathIgnorePatterns).toEqual(['b', 'a']);
     expect(options.setupFiles.sort()).toEqual([
       '/node_modules/a',
       '/node_modules/b',
       '/node_modules/regenerator-runtime/runtime',
     ]);
   });
+
+  test('merges with options and moduleNameMapper preset is overridden by options', () => {
+    /* eslint-disable sort-keys */
+    // prettier-ignore
```

Comment:
Thanks for the review. See my [comment above](https://github.com/facebook/jest/pull/3689#issuecomment-304869872) about the sort-keys rule.
I really think that a test for the correct handling of the order precedence of user and preset config should not only give a scenario that could imply that alphabetical order is the precedence of choice. Providing the object properties out of order makes the test more explicit and more resilient in my opinion. 
I'm happy to change my test to support the rule and remove the disable if you prefer the following of the style rules over a more resilient test. Let me know ;)